packages/protocol/lib/bytecode-foundry.ts

```diff
@@ -0,0 +1,201 @@
+/* eslint-disable max-classes-per-file: 0 */
+
+import { NULL_ADDRESS, trimLeading0x } from '@celo/base/lib/address'
+import { Artifact, LinkReferences } from '@celo/protocol/lib/compatibility/internal'
+import { keccak256, toHex } from 'viem'
+
+/*
+ * The Solidity compiler appends a Swarm Hash of compilation metadata to the end
+ * of bytecode. We find this hash based on the specification here:
+ * https://solidity.readthedocs.io/en/develop/metadata.html#encoding-of-the-metadata-hash-in-the-bytecode
+ */
+const CONTRACT_METADATA_REGEXPS = [
+  // 0.5.8
+  'a165627a7a72305820.{64}0029',
+  // 0.5.13
+  'a265627a7a72315820.{64}64736f6c6343.{6}0032',
+  // 0.8.19
+  'a264697066735822.{68}64736f6c6343.{6}0033'
+]
```

Comment:
Something to discuss in future -> but we might want to disable metadata trailing to allow for deterministic deployments across machines: https://github.com/celo-org/op-succinct/blob/mc01/deterministic-deployment/contracts/foundry.toml#L32